lib/coinbase/transaction.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+
+require_relative 'constants'
+require 'bigdecimal'
+require 'eth'
+require 'json'
+
+module Coinbase
+  # A representation of an onchain Transaction.
+  # Transactions should be constructed via higher level abstractions like Trade or Transfer.
+  class Transaction
+    # A representation of a Transaction status.
+    module Status
+      # The Transaction is awaiting being broadcast to the Network.
+      # At this point, transaction hashes may not yet be assigned.
+      PENDING = 'pending'
+
+      # The Transaction has been broadcast to the Network.
+      # At this point, at least the transaction hash should be assigned.
+      BROADCAST = 'broadcast'
+
+      # The Transaction is complete and has confirmed on the Network.
+      COMPLETE = 'complete'
+
+      # The Transaction has failed for some reason.
+      FAILED = 'failed'
+
+      # The states that are considered terminal on-chain.
+      TERMINAL_STATES = [COMPLETE, FAILED].freeze
+    end
```

Comment:
use symbols like in transfer.rb? Ref: https://github.com/coinbase/coinbase-sdk-ruby/blob/5040d0d7a000c24f95f234bd8a0d47f2090b6cb5/lib/coinbase/transfer.rb#L18

Reply:
We are removing Transfer::Status in the next PR.

We are returning strings from the API and I think we should just be using the frozen strings in this case

Reply:
All of our statuses are returned via strings and all comparisons should use these constants and not the symbol.

I think using the frozen string for comparison purposes is cleanest here!